src/test/java/teammates/common/datatransfer/attributes/NotificationAttributesTest.java

```diff
@@ -0,0 +1,269 @@
+                NotificationType.DEPRECATION, NotificationTargetUser.INSTRUCTOR,
+                "valid notification title", "valid notification message", false,
+                Instant.now(), Instant.now());
+        NotificationAttributes nfa = NotificationAttributes.valueOf(notification);
+        assertEquals(notification.getNotificationId(), nfa.getNotificationId());
+        assertEquals(notification.getStartTime(), nfa.getStartTime());
+        assertEquals(notification.getEndTime(), nfa.getEndTime());
+        assertEquals(notification.getType(), nfa.getType());
+        assertEquals(notification.getTargetUser(), nfa.getTargetUser());
+        assertEquals(notification.getTitle(), nfa.getTitle());
+        assertEquals(notification.getMessage(), nfa.getMessage());
+        assertEquals(notification.isShown(), nfa.isShown());
+        assertEquals(notification.getCreatedAt(), nfa.getCreatedAt());
+        assertEquals(notification.getUpdatedAt(), nfa.getUpdatedAt());
+    }
+
+    @Test
+    public void testBuilder_withNullArguments_shouldThrowException() {
+        assertThrows(AssertionError.class, () -> {
+            NotificationAttributes
+                    .builder(null)
+                    .build();
+        });
+
+        assertThrows(AssertionError.class, () -> {
+            NotificationAttributes
+                    .builder("notificationId")
+                    .withStartTime(null)
+                    .build();
+        });
+
+        assertThrows(AssertionError.class, () -> {
+            NotificationAttributes
+                    .builder("notificationId")
+                    .withEndTime(null)
+                    .build();
+        });
+
+        assertThrows(AssertionError.class, () -> {
+            NotificationAttributes
+                    .builder("notificationId")
+                    .withType(null)
+                    .build();
+        });
+
+        assertThrows(AssertionError.class, () -> {
+            NotificationAttributes
+                    .builder("notificationId")
+                    .withTargetUser(null)
+                    .build();
+        });
+
+        assertThrows(AssertionError.class, () -> {
+            NotificationAttributes
+                    .builder("notificationId")
+                    .withTitle(null)
+                    .build();
+        });
+
+        assertThrows(AssertionError.class, () -> {
+            NotificationAttributes
+                    .builder("notificationId")
+                    .withMessage(null)
+                    .build();
+        });
+    }
+
+    @Test
+    public void testBuilder_withTypicalData_shouldBuildCorrectAttributes() {
+        Notification notification = new Notification(STARTTIME_ONE, ENDTIME_ONE,
+                NotificationType.TIPS, NotificationTargetUser.GENERAL,
+                "Another tip for usage", "Here the message starts");
+        NotificationAttributes nfa = NotificationAttributes.valueOf(notification);
```

Comment:
For example this is a typical notification, we can simply call `generateTypicalNotificationAttributesObject()`